### PR TITLE
Adding docstring for IRHeader

### PR DESCRIPTION
**Pull request by maddyonline**
I used http://stackoverflow.com/a/20573384 as a guide to write docstring for namedtuple directly beneath it.

I am not sure if this documentation actually adds clarity or is in fact not required. Either way, creating a PR for any comments. (Feel free to close, if this is not required.)

@mli @zackchase @madjam @nswamy

### CODE REVIEW

File: python/mxnet/recordio.py [hunk 1]
```diff
@@ -166,6 +166,19 @@ def write_idx(self, idx, buf):
...
+Parameters
+----------
+flag : int
+    available for convenience, can be set arbitrarily
+label : float or an array of float
+    typically used to store label(s) for a record
+id: int
+    usually a unique id representing record
+id2: int
+    available for convenience, can be set arbitrarily
```
Comment (piiswrong) [inline python/mxnet/recordio.py]: I think this is the higher bits of id. Should be 0

Comment (maddyonline) [inline python/mxnet/recordio.py]: Fixed. Thanks. Let me know if a better comment is in order.

File: python/mxnet/recordio.py [hunk 2]
```diff
@@ -166,6 +166,19 @@ def write_idx(self, idx, buf):
 
 
 IRHeader = namedtuple('HEADER', ['flag', 'label', 'id', 'id2'])
+"""An alias for HEADER used for storing metadata (e.g. labels) that accompanies a record.
```
Comment (zackchase) [inline python/mxnet/recordio.py]: Right now the sentence runs on a bit. Please change to:

"An alias for HEADER. Used to store metadata (e.g. labels) accompanying a record."

File: python/mxnet/recordio.py [hunk 3]
```diff
@@ -166,6 +166,19 @@ def write_idx(self, idx, buf):
...
+See mxnet.recordio.pack and mxnet.recordio.pack_img for example uses.
+Parameters
+----------
+flag : int
+    available for convenience, can be set arbitrarily
```
Comment (zackchase) [inline python/mxnet/recordio.py]: Capitalize: "Available for convenience ..."

File: python/mxnet/recordio.py [hunk 4]
```diff
@@ -166,6 +166,19 @@ def write_idx(self, idx, buf):
...
+label : float or an array of float
+    typically used to store label(s) for a record
```
Comment (zackchase) [inline python/mxnet/recordio.py]: Capitalize "Typically"

Comment (zackchase) [inline python/mxnet/recordio.py]: End in period.

File: python/mxnet/recordio.py (hunk 3 above)
Comment (zackchase) [inline python/mxnet/recordio.py]: end in period.

File: python/mxnet/recordio.py [hunk 5]
```diff
@@ -166,6 +166,19 @@ def write_idx(self, idx, buf):
...
+id: int
+    usually a unique id representing record
```
Comment (zackchase) [inline python/mxnet/recordio.py]: -> "Usually a unique id representing record."

File: python/mxnet/recordio.py (hunk 1 above)
Comment (zackchase) [inline python/mxnet/recordio.py]: -> "Available for convenience, can be set arbitrarily."

**Comment by nswamy** [comment]
@maddyonline, Thanks for your contribution. could you please make the suggested changes so we could merge this?

**Comment by maddyonline** [comment]
@zackchase Thanks for suggestions. @nswamy Addressed comments, see https://github.com/dmlc/mxnet/pull/6114/commits/7decb10721d02a3aa24ff520eefaf1baa7331187

Let me know if you want me to squash my commits before merging.

File: python/mxnet/recordio.py [hunk 6]
```diff
@@ -272,6 +272,19 @@ def write_idx(self, idx, buf):
 
 
 IRHeader = namedtuple('HEADER', ['flag', 'label', 'id', 'id2'])
+"""An alias for HEADER. Used to store metadata (e.g. labels) accompanying a record.
+See mxnet.recordio.pack and mxnet.recordio.pack_img for example uses.
+Parameters
+----------
+flag : int
+    Available for convenience, can be set arbitrarily.
```
Comment (piiswrong) [inline python/mxnet/recordio.py]: I don't think this is true.

Comment (maddyonline) [inline python/mxnet/recordio.py]: What would be the correct documentation then?

Comment (piiswrong) [inline python/mxnet/recordio.py]: Actually I doesn't hurt for current implementation.

**Pull request merged**